### PR TITLE
[MLIR] Fix documentation generation for DLTI dialect.

### PR DESCRIPTION
**Pull request by AntonLydike**
This patch fixes the docs generation for the DLTI dialect to include attributes (CMake config pointed to the wrong tablegen file). It also fixes formatting in places so that proper markdown is generated.

Tagging @nhasabni @rengolin @ftynse

### CODE REVIEW

**Review by rengolin — APPROVED** [review]
thanks!

**Comment by AntonLydike** [comment]
@rengolin do I need another approval before merging? Just got commit access, so am happy to press the button, just wanted to check if it's ready.

**Comment by rengolin** [comment]
Technically you only need one approval, but it's usually good to wait a day to make sure there are no additional concerns. We can always do post-commit review if anyone finds something wrong after merge, and can either land a fix or revert for further investigations.

In this case, it should be safe to merge (no bots will break) and it's easy to fix later (another doc patch).

**Pull request merged**
